flutter/lib/providers/superformula_provider.dart

```diff
@@ -0,0 +1,227 @@
+//Flutter packages
+import 'dart:convert';
+//Third party packages
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+//My Packages
+import '../utils.dart';
+import '../values.dart' as values;
+
+enum DOWNLOAD_STATE{
+  DOWNLOADING,
+  DOWNLOADED,
+  ERROR,
+}
+
+class SuperFormulaProvider with ChangeNotifier{
+
+  //Object declarations
+
+  DOWNLOAD_STATE qrCodeDownloadingState = DOWNLOAD_STATE.DOWNLOADING;
+
+  String _seed = "";
+
+  String get seed{
+    return _seed;
+  }
+
+  //Network Functions
+
+  Dio dio = Dio(BaseOptions(
+    baseUrl: values.baseUrl,
+    connectTimeout: 8000,
+    receiveTimeout: 8000,
+  ));
+
+  Future getSeed({required bool firstTime,required BuildContext context}) async{
+
+    //Change state to downloading
+    qrCodeDownloadingState = DOWNLOAD_STATE.DOWNLOADING;
+
+    try {
+
+      //Execute the network call
+      var response = await dio.request(
+        "/seed",
+        options: Options(method: 'GET'),
+      );
+
+      //Hide the loader
+      qrCodeDownloadingState = DOWNLOAD_STATE.DOWNLOADED;
```

Comment:
**question:** does this actually immediately hide the loader in the views utilizing this provider?

Reply:
Yes, since it is a provider variable which the subscriber class listens to.